### PR TITLE
NEXUS-7924 nuget groups

### PR DESCRIPTION
**Pull request by mrprescott**
https://issues.sonatype.org/browse/NEXUS-7924

### CODE REVIEW

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java [hunk 1]
```diff
@@ -77,7 +78,7 @@ public boolean member(final Repository repository) {
   @Override
   @Guarded(by = STARTED)
   public List<Repository> members() {
-    List<Repository> members = Lists.newArrayListWithCapacity(memberNames.size());
+    List<Repository> members = new ArrayList<>(memberNames.size());
```
Comment (cstamas) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java]: Why this change?

Comment (mrprescott) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/group/GroupFacetImpl.java]: The javadoc for newArrayListWithCapacity says that it's deprecated as of Java 7, since the diamond syntax does the same job. Same with newArrayList(), newArrayList(E.. elements), and several other Lists methods.

**Comment by mrprescott** [comment]
http://bamboo.s/browse/NX3-OSSF125-1 :white_check_mark:

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java [hunk 2]
```diff
@@ -182,20 +190,51 @@ public String feed(final String base, final String operation, final Map<String,
...
+
+    if (result.getSkipLinkEntry() != null) {
+      xml.append("  <link rel=\"next\" href=\"").append(result.getBase()).append('/').append(result.getOperation());
+      xml.append("()?").append(ODataFeedUtils.skipLink(result.getSkipLinkEntry(), result.getQuery())).append("\"/>\n");
+    }
     return xml.append("</feed>").toString();
   }
 
+  private void setFeedQueryDefaults(final Map<String, String> query)
+  {// NEXUS-6822 Visual Studio doesn't send a sort order by default, leading to unusable results
```
Comment (kellyrob99) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java]: bad format here

**Comment by kellyrob99** [comment]
+1

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java (hunk 2 above)
Comment (mrprescott) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/NugetGalleryFacetImpl.java]: Fixed format.

File: plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java [hunk 3]
```diff
@@ -179,27 +195,30 @@ private URI nugetQuery(final String path, final Map<String, String> queryParams)
...
+        final Integer cachedCount = cache.get(key, remoteCall.build(repo, path));
+        counts.add(cachedCount);
+      }
+      catch (ExecutionException | UncheckedExecutionException e) {
+        log.warn("{} attempting to contact proxied repository {}.", e.getCause().getClass().getSimpleName(),
+            repo.getName());
+      }
+      catch (Exception e) {
+        log.warn("{} attempting to contact proxied repository {}.", e.getClass().getSimpleName(), repo.getName());
+      }
```
Comment (jdillon) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java]: both of these can eat exception detail, was this on purpose?

Comment (mrprescott) [inline plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/proxy/NugetProxyGalleryFacet.java]: Quite right, fixed.

**Comment by jdillon** [comment]
+1

**Comment by mrprescott** [comment]
Manually merged.
